common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -321,6 +321,10 @@ public String toString() {
       appendUnit(sb, rest, "microsecond");
     }
 
+    if (months == 0 && microseconds == 0) {
+      sb.append(" 0 microseconds");
```

Comment:
Note that `appendUnit` is designed to ignore `0` and this is more simpler (1 `append` call) than `appendUnit` (5 `append` calls)

Reply:
This seems fine; even just "0" seems reasonable as 0 seconds, microseconds, etc are all the same

Reply:
Thank you for review, @srowen .

Reply:
nit: How about this?
```
    } else if (months == 0) {
      sb.append(" 0 microseconds");
    }
```
In most cases, it seems microseconds is not zero, so I feel better to reduce #branches.
LGTM except one minor comment.

Reply:
Thank you for review and approval. Sure. I'll fix like that.